p2p/host/autorelay/autorelay.go

```diff
@@ -71,14 +73,15 @@ func NewAutoRelay(bhost *basic.BasicHost, discover discovery.Discoverer, router
 		router:     router,
 		addrsF:     bhost.AddrsFactory,
 		static:     static,
-		relays:     make(map[peer.ID]struct{}),
+		relays:     make(map[peer.ID]*circuitv2.Reservation),
 		disconnect: make(chan struct{}, 1),
 		status:     network.ReachabilityUnknown,
 	}
 	bhost.AddrsFactory = ar.hostAddrs
 	bhost.Network().Notify(ar)
 	ar.refCount.Add(1)
 	go ar.background(ctx)
+	go ar.refresh(ctx)
```

Comment:
Can we avoid starting a new go routine here by merging this into `background`?

Reply:
probably.

Reply:
done.